libs/types/src/cdf/election-results-reporting/convert.ts

```diff
@@ -0,0 +1,269 @@
+import { assertDefined, err, ok, Result } from '@votingworks/basics';
+import getDeepValue from 'lodash.get';
+import { Candidate, CandidateId } from '../../election';
+import { LanguageCode } from '../../language_code';
+import * as ResultsReporting from '.';
+import * as VxTabulation from '../../tabulation';
+import {
+  BallotMeasureContest,
+  CandidateContest,
+  CountItemType,
+  PartyContest,
+  RetentionContest,
+} from '.';
+import {
+  isBallotMeasureContest,
+  isCandidateContest,
+  isRetentionContest,
+} from './types';
+
+type CandidateNameRecord = Record<CandidateId, Candidate['name']>;
+
+function assertGetDeepValue(root: unknown, valuePath: string) {
+  return assertDefined(
+    getDeepValue(root, valuePath),
+    `Couldn't get value: ${valuePath}`
+  );
+}
+
+export interface LanguageStringQueryParams {
+  language?: LanguageCode;
+  content?: RegExp;
+}
+
+export function findLanguageString(
+  textEntries: readonly ResultsReporting.LanguageString[],
+  { language = LanguageCode.ENGLISH, content = /.*/ }: LanguageStringQueryParams
+): ResultsReporting.LanguageString | undefined {
+  function textContentFilter(entry: ResultsReporting.LanguageString): boolean {
+    return !!entry.Content.match(content);
+  }
+  function languageFilter(entry: ResultsReporting.LanguageString): boolean {
+    return entry.Language === language;
+  }
+
+  return textEntries.find(
+    (entry: ResultsReporting.LanguageString) =>
+      textContentFilter(entry) && languageFilter(entry)
+  );
+}
+
+export function findBallotMeasureSelectionWithContent(
+  content: RegExp,
+  ballotMeasureSelections: readonly ResultsReporting.BallotMeasureSelection[]
+): ResultsReporting.BallotMeasureSelection {
+  return assertDefined(
+    ballotMeasureSelections.find((selection) => {
+      const internationalizedText = selection.Selection;
+      return !!findLanguageString(internationalizedText.Text, {
+        content,
+      });
+    }),
+    `Could not find ballot measure selection with content "${content.toString()}"`
+  );
+}
+
+function findTotalVoteCounts(
+  voteCounts: readonly ResultsReporting.VoteCounts[]
+): number {
+  return assertDefined(
+    voteCounts.find((vc) => vc.Type === CountItemType.Total),
+    'Could not find total vote count'
+  ).Count;
+}
+
+function convertToYesNoContest(
+  contest:
+    | ResultsReporting.BallotMeasureContest
+    // RetentionContest inherits from BallotMeasureContest.
+    // It also has additional fields for candidate that we ignore.
+    | ResultsReporting.RetentionContest
+): VxTabulation.YesNoContestResults {
+  const contestSelections = assertDefined(
+    contest.ContestSelection
+  ) as ResultsReporting.BallotMeasureSelection[];
+  const yesOption = findBallotMeasureSelectionWithContent(
+    /yes/i,
+    contestSelections
+  );
+  const noOption = findBallotMeasureSelectionWithContent(
+    /no/i,
+    contestSelections
+  );
```

Comment:
I don't think there's necessarily a guarantee that these contest options will have "yes" or "no" in them. It might be safer to just use the order of the contest options (assign the first to yes, the second to no). There's not really any meaning to which one is assigned yes vs no in our internal schema.

Reply:
Is there a strong convention that yes is always first and no is always second? I'm really hesitant to enforce a convention like that with a file format that is so flexible.

By explicitly looking for "yes" and "no", and throwing if no matches are found, the worst that happens is election staff need to manually enter results.

By enforcing ordering, the worst that happens is we accidentally flip the results of a contest.

Reply:
Hmm I didn't think about it needing to match the existing contest options in the election definition. I think this is a tricky problem, because it's very possible to give the options labels that don't have yes/no in them. Maybe we do a hybrid approach where we look for yes/no to cover the common cases, but fall back to order if not? In my experience, it's common for option order to be pretty reliable, especially for ballot measures.

Reply:
Out of fear for an extreme edge case where we flip contest results, I support @kshen0's suggestion of looking for "yes" and "no" and, if they are not found, failing the import.

Reply:
Ok, I guess the mitigation if the contest options had labels without yes/no in them would be to edit the ERR file and replace those labels with yes/no, so I can get behind that.